dialyzer: Add no_extra_return and no_missing_return to warn_options type

gonzalobf opened the pull request:
When `extra_return` and `missing_return` were added in https://github.com/erlang/otp/pull/5214, the values weren't added to warn_options. This caused the following error when I try to skip any of the errors with `-dialyzer()`.

I used the same code as https://github.com/erlang/otp/pull/5214.
```erlang
-module(t).
-export([t/1, q/1]).

-dialyzer({no_missing_return, t/1}).
-spec t(integer()) -> odd.
t(N) when N rem 2 =/= 0 ->
    odd;
t(_) ->
    even.

-dialyzer({no_extra_return, q/1}).
-spec q(integer()) -> positive | negative | zero.
q(N) when N > 0 ->
    positive;
q(N) when N < 0 ->
    negative.
```

```
$ dialyzer t.erl -Wextra_return -Wmissing_return                                                                                                
  Checking whether the PLT /home/gonzalo/.cache/erlang/.dialyzer_plt is up-to-date... yes
  Proceeding with analysis...
dialyzer: Analysis failed with error:
Could not scan the following file(s):
/tmp/t.erl:4:2: unknown dialyzer warning option: no_extra_return
/tmp/t.erl:11:2: unknown dialyzer warning option: no_missing_return
```

`-Wno_extra_return` and `-Wno_missing_return` are defined at the end of the section in this  documentation page https://www.erlang.org/doc/man/dialyzer.html#using-dialyzer-from-the-command-line so I think they are wanted as warning option too.

I realised about the missing options when I was trying to add new dialyzer options  `-Wextra_return` and `Wmissing_return`, and I just wanted to fix a few warnings and skip a few others.

jhogberg commented:
Thanks for the PR, it looks good and I'd like to merge it. Can you rebase it to `maint` so we can include it in OTP 25.1? :)

gonzalobf commented:
@jhogberg Thanks, I rebased `maint` and changed the base branch to `maint` too. Hopefully, everything should be fined. Let me know if you need to do something else.

jhogberg commented:
Merged, thanks again for the PR!

gonzalobf commented:
@jhogberg Great! Thanks to you too!
